src/PowerShellEditorServices/Services/Analysis/AnalysisService.cs

```diff
@@ -941,43 +415,75 @@ private static DiagnosticSeverity MapDiagnosticSeverity(ScriptFileMarkerLevel ma
+        {
+            return new Hashtable {
+                { "Rules", new Hashtable {
+                    { "PSProvideCommentHelp", new Hashtable {
+                        { "Enable", true },
+                        { "ExportedOnly", false },
+                        { "BlockComment", forBlockComment },
+                        { "VSCodeSnippetCorrection", true },
+                        { "Placement", helpLocation }}
+                    }}
+                }};
+        }
 
-                case ScriptFileMarkerLevel.Information:
-                    return DiagnosticSeverity.Information;
+        #region IDisposable Support
+        private bool disposedValue = false; // To detect redundant calls
+
+        protected virtual void Dispose(bool disposing)
+        {
+            if (!disposedValue)
+            {
+                if (disposing)
+                {
+                    if (_analysisEngine.IsValueCreated) { _analysisEngine.Value.Dispose(); }
+                    _diagnosticsCancellationTokenSource?.Dispose();
+                }
 
-                default:
-                    return DiagnosticSeverity.Error;
+                disposedValue = true;
             }
         }
-    }
 
-    /// <summary>
-    /// Class to catch known failure modes for starting the AnalysisService.
-    /// </summary>
-    public class AnalysisServiceLoadException : Exception
-    {
-        /// <summary>
-        /// Instantiate an AnalysisService error based on a simple message.
-        /// </summary>
-        /// <param name="message">The message to display to the user detailing the error.</param>
-        public AnalysisServiceLoadException(string message)
-            : base(message)
+        // This code added to correctly implement the disposable pattern.
+        public void Dispose()
         {
+            // Do not change this code. Put cleanup code in Dispose(bool disposing) above.
+            Dispose(true);
         }
+        #endregion
 
         /// <summary>
-        /// Instantiate an AnalysisService error based on another error that occurred internally.
+        /// Tracks corrections suggested by PSSA for a given file,
+        /// so that after a diagnostics request has fired,
+        /// a code action request can look up that file,
+        /// wait for analysis to finish if needed,
+        /// and then fetch the corrections set in the table entry by PSSA.
         /// </summary>
-        /// <param name="message">The message to display to the user detailing the error.</param>
-        /// <param name="innerException">The inner exception that occurred to trigger this error.</param>
-        public AnalysisServiceLoadException(string message, Exception innerException)
-            : base(message, innerException)
+        private class CorrectionTableEntry
         {
+            public static CorrectionTableEntry CreateForFile(ScriptFile file)
```

Comment:
![Codacy](https://app.codacy.com/assets/images/favicon.png) Issue found: [Remove this unused method parameter 'file'.](https://app.codacy.com/manual/TylerLeonhardt/PowerShellEditorServices/pullRequest?prid=4961943)